features/cellular/framework/API/CellularDevice.h

```diff
@@ -37,6 +44,23 @@ namespace mbed {
  */
 class CellularDevice {
 public:
+
+    /** Return singleton instance of CellularDevice.
+     *
+     */
+    static CellularDevice *get_default_instance();
+
+    /** Get event queue that can be chained to main event queue. EventQueue is created in get_instance().
+     *  @return event queue
+     */
+    events::EventQueue &get_queue();
+
+protected:
+    // don't allow creating this class in any other way than get_default_instance()
+    CellularDevice();
```

Comment:
There's no inherent reason to block manual if people know exactly what they're doing - what modem(s) they're using and on what pins etc.

The "get_default_instance" is supposed to be a helper for people who don't know what they're doing, and just want a target-appropriate single default, but doesn't exclude manual creation.

`EthInterface::get_default_instance` or `EthernetInterface` does not preclude `MyCustomEthernet(PIN1,PIN2)` or `EthernetInterface(ParticularMAC)`.